### ISSUE TITLE
Agent does not send any data after throttling event

### ISSUE THREAD

**Issue by im-dim** [SumoLogic/logstash-output-sumologic#71]
Hello,

We had a situation when output was completely blocked after hitting a throttling event on Sumo side. The data started to flow only after bouncing pipeline. The logs below are the last messages in the log. Nothing else was logged till the bounce. The issue happened in four different data centers at the same time (same target error).

[2022-01-31T06:45:44,647][ERROR][logstash.outputs.sumologic][sumo_out] request rejected {:token=>64, :code=>429, :headers=>{"X-Sumo-Client"=>"logstash-output-sumologic", "X-Sumo-Category"=>"ca/elastiflow", "X-Sumo-Host"=>"my.domain", "X-Sumo-Name"=>"elastiflow", "Content-Type"=>"text/plain"}, :contet=>"{\"event\":{\"type\":\"net"}
[2022-01-31T06:45:45,713][ERROR][logstash.outputs.sumologic][sumo_out] request rejected {:token=>6, :code=>429, :headers=>{"X-Sumo-Client"=>"logstash-output-sumologic", "X-Sumo-Category"=>"ca/elastiflow", "X-Sumo-Host"=>"my.domain", "X-Sumo-Name"=>"elastiflow", "Content-Type"=>"text/plain"}, :contet=>"{\"event\":{\"type\":\"net"}

Thank you.

**Comment by sumo-drosiek** [comment]
@im-dim Sorry for late response. Did you observed the following log?
```
[INFO ] 2022-02-23 08:28:06.236 [pool-37-thread-1] sumologic - requeue message {:after=>30, :queue_size=>0, :content_size=>71, :content=>"2022-02-23T08:26:06.0", :headers=>{"X-Sumo-Client"=>"logstash-output-sumologic", "X-Sumo-Category"=>"Logstash", "X-Sumo-Host"=>"logstash-output-sumologic", "X-Sumo-Name"=>"logstash-output-sumologic", "Content-Type"=>"text/plain"}}
```

**Comment by sumo-drosiek** [comment]
I reproduced the issue and can explain what happens.

1. We are using queue to store data which has not been send due to errors and so on.
1. If we fail to send, [we push data on the end of queue](https://github.com/SumoLogic/logstash-output-sumologic/blob/main/lib/logstash/outputs/sumologic/sender.rb#L145)
1. We are using workers to dequeue, but in case worker fails to send it try to push data on the end of queue.
1. If queue is full workers are waiting for other workers to take data from queue so there will be space for their batch, but if all workers are waiting, there is no more worker to take data from queue and we are deadlocked.

**Comment by sumo-drosiek** [comment]
@im-dim We released `1.4.1` just now, which should resolve the issue
